tests/integration/test_reactive.py

```diff
@@ -0,0 +1,131 @@
+#  Copyright 2024 Canonical Ltd.
+#  See LICENSE file for licensing details.
+import json
+import random
+import secrets
+
+import pytest
+from juju.application import Application
+from juju.model import Model
+from juju.unit import Unit
+from kombu import Connection
+from pytest_operator.plugin import OpsTest
+
+from reactive.job import JobDetails
+from reactive.runner_manager import REACTIVE_RUNNER_LOG_PATH
+from tests.integration.helpers.common import get_file_content, reconcile
+
+FAKE_URL = "http://example.com"
+
+
+@pytest.mark.openstack
+async def test_reactive_mode_consumes_jobs(ops_test: OpsTest, app_for_reactive: Application):
+    unit = app_for_reactive.units[0]
+    mongodb_uri = await _get_mongodb_uri_from_integration_data(ops_test, unit)
+    if not mongodb_uri:
+        mongodb_uri = await _get_mongodb_uri_from_secrets(ops_test, unit.model)
```

Comment:
I think it will be kept secret if it's Juju 3, or it will be related to data if it's Juju 2.

https://github.com/canonical/mongodb-operator/blob/b33d036173f47c68823e08a9f03189dc534d38dc/lib/charms/data_platform_libs/v0/data_interfaces.py#L827-L831